### PR TITLE
SSCS-11743-Action-postponment-update-case-state

### PR DESCRIPTION
**Pull request by StevensonSi**
**Before creating a pull request make sure that:**

- [ x] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ x] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

[### JIRA link (if applicable) ###
](https://tools.hmcts.net/jira/browse/SSCS-11743)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java [hunk 1]
```diff
@@ -120,6 +121,13 @@ private void grantPostponement(SscsCaseData sscsCaseData, PreSubmitCallbackRespo
         log.info("Action postponement request: postponement listingOption {} mapped to Event {} for case {}",
             listingOption, postponementEventType, sscsCaseData.getCcdCaseId());
 
+
+        if (listingOption.equalsIgnoreCase("readytolist")) {
+            sscsCaseData.setState(State.READY_TO_LIST);
+        } else if (listingOption.equalsIgnoreCase("notlistable")) {
+            sscsCaseData.setState(State.NOT_LISTABLE);
```
Comment (nickhill111) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandler.java:128]: you can simplify this with:
sscsCaseData.setState(State.valueOf(listingOption));
Doing this should also sort out your Code coverage (check sonarcube)

**Review by nickhill111 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java [hunk 2]
```diff
@@ -188,6 +188,7 @@ public void givenARefusedPostponement_thenClearReviewStateAndReferralReasonAndFl
 
     @Test
     public void givenAGrantedPostponement_thenClearReviewStateAndReferralReasonAndFlagAndAddNoteAndDwpStateAndDecisionDocAdded() {
+        sscsCaseData.getPostponementRequest().setListingOption("readyToList");
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java:191]: Please save "readyToList" as a constant, or use State.READY_TO_LIST.getType(), and replace all usages in the test class

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java [hunk 3]
```diff
@@ -212,6 +214,28 @@ public void givenAGrantedPostponement_shouldSendCancellation() {
...
+        PreSubmitCallbackResponse<SscsCaseData> response =
+                handler.handle(ABOUT_TO_SUBMIT, callback, USER_AUTHORISATION);
+
+        assertThat(response.getData().getState()).isEqualTo(State.READY_TO_LIST);
+    }
+
+    @Test
+    public void givenAGrantedPostponement_shouldUpdateCaseStateToNotListable() {
+        sscsCaseData.setAppeal(Appeal.builder().hearingOptions(HearingOptions.builder().build()).build());
+        sscsCaseData.getPostponementRequest().setListingOption("notListable");
```
Comment (amoslv1) [inline src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/actionpostponementrequest/ActionPostponementRequestAboutToSubmitHandlerTest.java:231]: Can do same here, constant or getType

**Review by amoslv1 — CHANGES_REQUESTED** [review]
replace string literals in the test

**Review by nickhill111 — DISMISSED** [review]

**Review by amoslv1 — APPROVED** [review]

**Review by jarekPierchala — APPROVED** [review]

**Review by nickhill111 — APPROVED** [review]

**Pull request merged**